### DISCUSSION TITLE
Flask-SocketIO and Flask-Login together with socket.io-client

### DISCUSSION THREAD

**Discussion by McToel** [miguelgrinberg/Flask-SocketIO#1692] [General]
I want to use Flask-SocketIO and Flask-Login together with socket.io-client. However, the current_user in SocketIO functions is always anonymous. I think the reason for this is, that socket.io-client does not send the session cookie.
Any Ideas on how to solve this?

**Comment by miguelgrinberg** [comment]
@McToel the socket.io client is connected over WebSocket. There is no support for sending or updating cookies.

Have you read the documentation? In particular the sections that discuss how the user session works under Socket.IO and how to use Flask-Login? There is also a code example in this repository called `sessions.py` that shows how to use Flask-Login.

**Reply by McToel** [reply]
Now it works, but I really don't get how, because I changed nothing. The one thing I noticed, that it now shows the session cookie being sent to the server in the Firefox developer panel, which didn't happen before (for the initial connection with the WebSocket and all the time if I use polling). And yes, I did read the docs on socket.io and logins.
Thanks for your help!
